code/modules/food_and_drinks/kitchen_machinery/griddle.dm

```diff
@@ -63,6 +63,37 @@
 
 
 /obj/machinery/griddle/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/tray))
+		var/datum/component/storage/tray_storage = I.GetComponent(/datum/component/storage)
+		if(!I.contents.len && griddled_objects.len > 0) //if the griddle has things on it and the tray is empty, put what you can into the tray
+			for(var/obj/item/grilled as anything in griddled_objects)
+				if(!griddled_objects.len)
+					return
+				if(I.contents.len >= tray_storage.max_items)
+					to_chat(user, span_notice("[I] can't fit more items!")) //tray is full
+					return
+				if(SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					ItemRemovedFromGrill(grilled)
+				else if(!SEND_SIGNAL(I, COMSIG_TRY_STORAGE_INSERT, grilled, user))
+					to_chat(user, span_notice("[I] can't hold [grilled]!")) //if an item cant enter the tray, dont touch it
```

Comment:
"Is there a better way of doing this? these signals smell horrible, but it may just be a cost of the storage component being the way it is."

unfortunately no, ive tried just doing various "move" procs that i know work in other instances such as moving items from a console to your hands, but in testing, every single one would just delete the items rather than moving them, nothing works to move from the griddle to the tray except this signal, unsure why, probably because griddles arent actually a storage so has nothing to move from?


as for sending it twice, because theres two reasons the loop can fail to pickup an item: either because the item just cant be picked up, or its empty, so i cant do an "else" for the bottom one because theres other "else" that might be called and i dont want it to say "I cant be grilled!" if thats not actually why the else is being called